org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/generators/AttributeDeclaration.java

```diff
@@ -31,4 +56,144 @@ public AttributeDeclaration(String name) {
 	public String getName() {
 		return name;
 	}
+
+	/**
+	 * Increment the DOM attributes occurrence.
+	 */
+	public void incrementOccurrences() {
+		this.occurrences++;
+	}
+
+	public void addValue(String value) {
+		if (!values.contains(value)) {
+			// We haven't seen this attribute value before
+			values.add(value);
+
+			// Check if attribute value is a valid name
+			if (allNames && !XMLChar.isValidName(value)) {
+				allNames = false;
+			}
+			// Check if attribute value is a valid NMTOKEN
+			if (allNMTOKENs && !XMLChar.isValidNmtoken(value)) {
+				allNMTOKENs = false;
+			}
+			if (dataType == DataType.UNKWOWN) {
+				dataType = getDataType(value);

```

Comment:
If we have this XML:
```
<note>
    <C version="1.2"></C>
    <C version="hi"></C>
</note>
```

This generates:
```
<xs:attribute name="version" type="xs:decimal" use="required" />
```

Which creates an error because `"hi"` is not a decimal

Reply:
fixed